benchmark_runner/common/oc/oc.py

```diff
@@ -667,90 +592,230 @@ def wait_for_pod_completed(self, label: str, workload: str = '', label_uuid: boo
+        """
+        current_wait_time = 0
+        while current_wait_time <= timeout:
+            if self.vm_exists(vm_name=vm_name, namespace=namespace):
+                return True
+            # sleep for x seconds
+            time.sleep(OC.SLEEP_TIME)
+            current_wait_time += OC.SLEEP_TIME
+        raise VMNotCreateTimeout(vm_name)
+
+    @typechecked
+    @logger_time_stamp
+    def create_vm_sync(self, yaml: str, vm_name: str, namespace: str = environment_variables.environment_variables_dict['namespace'], timeout: int = int(environment_variables.environment_variables_dict['timeout'])):
+        """
+        This method create vm yaml in async
+        :param timeout:
+        :param vm_name:
+        :param yaml:
         :return:
         """
-        if self.pod_exists(pod_name, namespace):
-            try:
-                if namespace != '':
-                    namespace = f'-n {namespace}'
-                self.run(f'{self.__cli} delete pod {namespace} {pod_name} timeout={timeout}')
-            except Exception as err:
-                raise PodTerminateTimeout(pod_name)
+        self._create_async(yaml)
+        return self.wait_for_vm_create(vm_name=vm_name, namespace=namespace, timeout=timeout)
 
     @typechecked
     @logger_time_stamp
-    def wait_for_pod_ready(self, pod_name: str,
-                           namespace: str = environment_variables.environment_variables_dict['namespace'],
-                           timeout: int = int(environment_variables.environment_variables_dict['timeout'])):
+    def delete_vm_sync(self, yaml: str, vm_name: str,
+                       namespace: str = environment_variables.environment_variables_dict['namespace'],
+                       timeout: int = int(environment_variables.environment_variables_dict['timeout'])):
         """
-        Wait for a pod to be ready and running
-        :param pod_name:
+        This method delete vm yaml in async, only if exist and return false if not exist
         :param namespace:
         :param timeout:
-        :return: True if getting pod name or raise PodNameError
+        :param vm_name:
+        :param yaml:
+        :return:
+        """
+        if self.vm_exists(vm_name=vm_name, namespace=namespace):
+            self._delete_async(yaml)
+            return self.wait_for_vm_terminate(vm_name=vm_name, namespace=namespace, timeout=timeout)
+        else:
+            return False
+
+    @logger_time_stamp
+    def wait_for_vm_completed(self, workload: str = '', vm_name: str = '',
+                              namespace: str = environment_variables.environment_variables_dict['namespace'],
+                              timeout: int = int(environment_variables.environment_variables_dict['timeout'])):
+        """
+        This method wait to pod to be completed
+        :param vm_name: The vm name
+        :param workload:
+        :param namespace:
+        :return:
```

Comment:
This returns a value in some cases but no value in others.

Reply:
The `:return:` in the pydoc string should indicate that it returns a value, but given that it always either returns or raises an exception, returning nothing would make more sense.